lib/web_ui/lib/src/engine/window.dart

```diff
@@ -155,7 +209,9 @@ class EngineWindow extends ui.Window {
   /// Engine code should use this method instead of the callback directly.
   /// Otherwise zones won't work properly.
   void invokeOnMetricsChanged() {
-    _invoke(_onMetricsChanged, _onMetricsChangedZone);
+    if (window._onMetricsChanged != null) {
```

Comment:
[Sorry for the drive-by, but I noticed this when I merged my workspace with your change.]

This isn't necessary: `_invoke` already checks for null.

Also, even if it didn't, you should just use `_onMetricsChanged` here, since the `window` singleton might not always be identical to `this` in the future.